vm/src/builtins/set.rs

```diff
@@ -658,7 +658,11 @@ impl PySet {
     }
 
     #[pymethod(magic)]
-    fn iand(zelf: PyRef<Self>, set: AnySet, vm: &VirtualMachine) -> PyResult<PyRef<Self>> {
+    fn iand(zelf: PyRef<Self>, set: PyObjectRef, vm: &VirtualMachine) -> PyResult<PyRef<Self>> {
+        if zelf.get_id() == set.get_id() {
+            return Ok(zelf);
+        }
```

Comment:
`intersection_update` also needs to have a check like this. As `iand` and `intersection_update` share the same `PySetInner::intersection_update`, we can do better 😊